### PR TITLE
Fix dataset operation highlighting in Safari and dataset error route

### PR DESCRIPTION
**Pull request by guerler**
Addition to #14487 to fix the issue in Safari. This PR also fixes a recently introduced wrong route for dataset error links, probably due to a difficult merge. The dataset visualizations route is also not working and should be followed up with.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by ElectronicBlueberry** [comment]
This forces all the buttons to render as `<a>` elements, which has unexpected hover and middle mouse button interactions.
I'd prefer if we could find a way to fix this with css.

**Comment by guerler** [comment]
@ElectronicBlueberry thank you for looking into this again, feel free to take another stab at it, I just wanted to fix the inconsistent highlighting i.e. some buttons show a background and others do not although next to each other. Generally the PR just adds empty `hrefs`, the buttons are already registered as `variant: link`, since links are rendered slightly differently if they contain an `href` vs if they do not.

**Comment by ElectronicBlueberry** [comment]
`<b-button variant="link">` renders to `<button type="button" class="btn btn-link">`

`<b-button variant="link" href="#">` renders to `<a role="button" tabindex="0" href="#" target="_self" class="btn btn-link">`

Adding `href` changes the rendered element type from button to link, which is no longer semantically correct, as these buttons are not links, and using them as links (eg. middle mouse clicking) will not give expected results.

I can unfortunately not look into this issue myself, as I do not have access to mac hardware.

My guess is that safari for some reason ignores line 1006 of `base.css`. Either due to some value issue with `transparent !important`, or due to a selector issue. I came to this conclusion, since by removing the line I can achieve the same result as your screenshot. I cannot confirm if this is the actual cause, but maybe it can give you a starting point.

**Comment by guerler** [comment]
I agree, it is probably a selector issue which was recently introduced, but we need to make sure that future changes to the UI styling do not break Safari since it worked consistently for a decade.

File: client/src/style/scss/base.scss [hunk 1]
```diff
@@ -999,10 +999,7 @@ a.action-button {
...
 
 .content-buttons {
-    .btn:hover,
-    .btn:active,
-    .btn:focus,
-    .btn:focus-visible {
+    .btn {
         background-color: transparent !important;
         border-color: transparent !important;
         box-shadow: none !important;
```
Comment (ElectronicBlueberry) [inline client/src/style/scss/base.scss]: Just discovered that this removes active highlighting. Seems to be introduced by an earlier PR. Removing it restores highlighting for me, without reintroducing style discrepancies. Could you check if this is also the case on Safari?
```suggestion
```

**Review by ElectronicBlueberry — APPROVED** [review]

**Pull request merged**